docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,119 @@
-`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+```mysql
+CREATE OR REPLACE 
+  VIEW PopularBooks AS 
+    SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1
+```  
 
 Once a view is created, you can use it as you query any table in a `SELECT` statement. For example, to list all the popular book titles ordered by an author, you could write:  
-`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+```mysql
+SELECT Author, Title FROM PopularBooks ORDER BY Author
+```
 
 In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
 
 ## Proposal
-I implement the "`CREATE OR REPLACE`" command to create a view with the `SELECT` statement. Currently, I reuse "`DROP TABLE`" to drop a view, and "`SHOW FULL TABLES`" can also display the view list with different table types. The "`SHOW CREATE TABlE`" command can also regenerate a view create command.
-
+This proposal is prepared to implement basic VIEW feature, which contains "CREATE OR REPLACE VIEW", "SELECT FROM VIEW", "DROP VIEW" and "SHOW TABLE STATUS". All other unimplemented feature will list as compatibility and discuss later.
+We introduce `ViewInfo` to store the metadata for view and add an attribute `*ViewInfo` which named `View` to TableInfo. If `TableInfo.ViewInfo` != nil, then this `TableInfo` is a base table, else this `TableInfo` is a view.:
+```
+type ViewInfo struct {
+	Algorithm   ViewAlgorithm    `json:"view_algorithm"`
+	Definer     UserIdentity     `json:"view_definer"`  
+	Security    ViewSecurity     `json:"view_security""`
+	SelectStmt  string           `json:"view_select"`
+	CheckOption ViewCheckOption  `json:"view_checkoption"`
+	Cols        []model.CIStr    `json:"view_cols"`
+}
+```
+* [Algorithm](https://dev.mysql.com/doc/refman/5.7/en/view-algorithms.html)  
+    The view SQL AlGORITHM characteristic. The value is one of UNDEFINED、MERGE OR TEMPTABLE, if no ALGORITHM clause is present, UNDEFINED is the default algorithm.
+    We will implement Algorithm=MERGE only now.
+* [Definer](https://dev.mysql.com/doc/refman/5.7/en/create-view.html)  
+    The account of the user who created the view, in 'user_name'@'host_name' format.
+* [Security](https://dev.mysql.com/doc/refman/5.7/en/create-view.html)  
+    The view SQL SECURITY characteristic. The value is one of DEFINER or INVOKER.
+* [CheckOption](https://dev.mysql.com/doc/refman/5.7/en/view-check-option.html)  
+    The WITH CHECK OPTION clause can be given for an updatable view to prevent inserts to rows for which the WHERE clause in the select_statement is not true. It also prevents updates to rows for which the WHERE clause is true but the update would cause it to be not true (in other words, it prevents visible rows from being updated to nonvisible rows).  
```

Comment:
```suggestion
    The WITH CHECK OPTION clause can be given to an updatable view to prevent inserts to rows for which the WHERE clause in the select_statement is not true. It also prevents updates to rows for which the WHERE clause is true but the update would cause it to be not true (in other words, it prevents visible rows from being updated to nonvisible rows).  
```